source/grid/grid_in.cc

```diff
@@ -1451,14 +1452,90 @@ GridIn<dim, spacedim>::read_msh(std::istream &in)
           in >> line;
         }
 
-      // if the next block is of kind $Entities, ignore it
+      // if the next block is of kind $Entities, parse it
       if (line == "$Entities")
         {
-          do
+          unsigned long n_points, n_curves, n_surfaces, n_volumes, n_physicals;
+          int           tag, physical_tag;
+          double        box_min_x, box_min_y, box_min_z, box_max_x, box_max_y,
+            box_max_z;
```

Comment:
Can you move the declaration of `tag`...`box_max_z` into each of the brace-enclosed blocks below? This duplicates their declarations, but at least one never has to wonder whether the state of a variable is carried from one block to another.